### PR TITLE
[MXNET-532] Clarify documentation of save_parameters(), load_parameters()

### PR DESCRIPTION
**Pull request by ThomasDelteil**
Update documentation to clarify the way of saving / loading parameters with Gluon, following the issue raised here: https://github.com/apache/incubator-mxnet/issues/11091

@szha @marcoabreu

### CODE REVIEW

File: python/mxnet/gluon/block.py [hunk 1]
```diff
@@ -309,6 +309,17 @@ def _collect_params_with_prefix(self, prefix=''):
...
     def save_params(self, filename):
         """Save parameters to file.
+        This function is to be used to save parameters of a Gluon model, note that
+        the saved parameters are not meant to be loaded in a different language binding for now.
+        Saving parameters using `.save_params()` is different than
+        `.collect_params().save()`, which is a deprecated way to save parameters of a model
+        and should be avoided.
+        If your model is hybridizable and you want to export a serialized version of the
+        structure of the model as well as its parameters please refer to
+        `HybridBlock.export()`. Such model can then be loaded back in any language binding
```
Comment (anirudh2290) [inline python/mxnet/gluon/block.py]: can we change this to :py:meth:`HybridBlock.export`

File: python/mxnet/gluon/block.py [hunk 2]
```diff
@@ -309,6 +309,17 @@ def _collect_params_with_prefix(self, prefix=''):
...
         """Save parameters to file.
+        This function is to be used to save parameters of a Gluon model, note that
+        the saved parameters are not meant to be loaded in a different language binding for now.
+        Saving parameters using `.save_params()` is different than
+        `.collect_params().save()`, which is a deprecated way to save parameters of a model
+        and should be avoided.
+        If your model is hybridizable and you want to export a serialized version of the
+        structure of the model as well as its parameters please refer to
+        :py:meth:`HybridBlock.export`. Such model can then be loaded back in any language binding
+        or even in Gluon using a `SymbolBlock`.
```
Comment (anirudh2290) [inline python/mxnet/gluon/block.py]: nit: also change to :py:class:`SymbolBlock`

**Comment by piiswrong** [comment]
These functions are going to be deprecated. I don't think we should add so much doc to them. A deprecation message is enough

**Comment by ThomasDelteil** [comment]
I think these docs should be ported to `save_parameters` and `load_parameters`

File: python/mxnet/gluon/block.py [hunk 3]
```diff
@@ -309,6 +309,17 @@ def _collect_params_with_prefix(self, prefix=''):
 
     def save_params(self, filename):
         """Save parameters to file.
+        This function is to be used to save parameters of a Gluon model, note that
+        the saved parameters are not meant to be loaded in a different language binding for now.
+        Saving parameters using `.save_params()` is different than
+        `.collect_params().save()`, which is a deprecated way to save parameters of a model
+        and should be avoided.
```
[the remark this answers is not in the document]
Comment (ThomasDelteil) [reply python/mxnet/gluon/block.py]: no I don't agree. I think the documentation here is the one that should be moved to save_parameters and load_parameters.

```
Saving parameters using `.save_params()` is different than
+        `.collect_params().save()`, which is a deprecated way to save parameters of a model
+        and should be avoided.
```
->
```
Saving parameters using `.save_parameters()` is different than
+        `.collect_params().save()` and .save_params() which are deprecated ways to save parameters of a model
+        and should be avoided.
```
Once the other PR is merged

[the remark this answers is not in the document]
Comment (ThomasDelteil) [reply python/mxnet/gluon/block.py]: But as it stands against current master and current save_params and load_params, this documentation is valid

[the remark this answers is not in the document]
Comment (anirudh2290) [reply python/mxnet/gluon/block.py]: yep you have a point. lets wait for https://github.com/apache/incubator-mxnet/pull/11127 and then make the necessary changes as part of this PR.

[the remark this answers is not in the document]
Comment (ThomasDelteil) [reply python/mxnet/gluon/block.py]: ok will wait for the other PR to be merged...

[the remark this answers is not in the document]
Comment (anirudh2290) [reply python/mxnet/gluon/block.py]: @ThomasDelteil #11127 is merged. Can you make the necessary changes.

**Pull request merged**